### PR TITLE
Improve RGS documentation

### PR DESCRIPTION
**Pull request by tnull**
Previously the `error` and `processing` modules have been `pub`, leading to them being included in the docs, although (almost) empty. Instead, we now `pub use` the `GraphSyncError` directly, cleaning up the docs.

Since I found they could be improved a bit, I also went over the crate-level docs and restructured them slightly.

### CODE REVIEW

File: lightning-rapid-gossip-sync/src/lib.rs [hunk 1]
```diff
@@ -10,24 +10,40 @@
...
+//! can be found [here](https://github.com/lightningdevkit/rapid-gossip-sync-server).
 //!
-//! An example server request could look as simple as the following. Note that the first ever rapid
-//! sync should use `0` for `last_sync_timestamp`:
+//! The primary benefit of this syncing mechanism is that it allows a low-powered client to offload
+//! the validation of gossip signatures to a trusted server. This enables the client to privately
+//! calculate routes for payments, and to do so much faster than requiring a full peer-to-peer
+//! gossip sync to complete.
+//!
+//! The reason the rapid sync server requires trust is that it could provide bogus data, though, at
```
Comment (tnull) [inline lightning-rapid-gossip-sync/src/lib.rs:27]: Not sure we want to leave this paragraph in, seems a bit out-of-place in a technical documentation.

Comment (tnull) [reply lightning-rapid-gossip-sync/src/lib.rs]: Removed the paragraph.

File: lightning-rapid-gossip-sync/src/lib.rs [hunk 2]
```diff
@@ -10,24 +10,40 @@
...
+//! worst, all that would result in is a fake network topology, which wouldn't enable the server to
+//! steal or siphon off funds. It could, however, reduce the client's privacy by forcing all
+//! payments to be routed via channels the server controls.
+//!
+//! The server calculates its repsonse on the basis of a client-provided `latest_seen` timestamp,
+//! i.e., it will return all rapid gossip sync data it has seen after the given timestamp.
+//!
+//! # Getting Started
+//! Firstly, the data needs to be retrieved from the server. An examplary server request could look
```
[the remark this answers is not in the document]
Comment (tnull) [reply lightning-rapid-gossip-sync/src/lib.rs]: Ha, right! Updated!

File: lightning-rapid-gossip-sync/src/lib.rs [hunk 3]
```diff
@@ -10,24 +10,36 @@
...
 //!
-//! An example server request could look as simple as the following. Note that the first ever rapid
-//! sync should use `0` for `last_sync_timestamp`:
+//! The primary benefit of this syncing mechanism is that it allows a low-powered client to offload
+//! the validation of gossip signatures to a trusted server. This enables the client to privately
+//! calculate routes for payments, and to do so much faster than requiring a full peer-to-peer
+//! gossip sync to complete.
+//!
+//! The server calculates its repsonse on the basis of a client-provided `latest_seen` timestamp,
+//! i.e., it will return all rapid gossip sync data it has seen after the given timestamp.
```
Comment (arik-so) [inline lightning-rapid-gossip-sync/src/lib.rs:28]: this phrasing is a bit more ambiguous than it was before, what with the timestamp being client-provided, and the snippet "data it has seen." It's not as clear that there, "it" is referring to the server.

File: lightning-rapid-gossip-sync/src/lib.rs [hunk 4]
```diff
@@ -10,24 +10,36 @@
...
+//! The rapid gossip sync server will provide a compressed response containing differential gossip
+//! data. The gossip data is formatted compactly, omitting signatures and opportunistically
+//! incremental where previous channel updates are known. This mechanism is enabled when the
+//! timestamp of the last known channel update is communicated. A reference server implementation
+//! can be found [on Github](https://github.com/lightningdevkit/rapid-gossip-sync-server).
 //!
-//! An example server request could look as simple as the following. Note that the first ever rapid
-//! sync should use `0` for `last_sync_timestamp`:
+//! The primary benefit of this syncing mechanism is that it allows a low-powered client to offload
+//! the validation of gossip signatures to a trusted server. This enables the client to privately
```
[the remark this answers is not in the document]
Comment (tnull) [reply lightning-rapid-gossip-sync/src/lib.rs:23]: Semi-done it is!

File: lightning-rapid-gossip-sync/src/lib.rs [hunk 5]
```diff
@@ -10,24 +10,36 @@
...
+//! calculate routes for payments, and to do so much faster than requiring a full peer-to-peer
+//! gossip sync to complete.
+//!
+//! The server calculates its repsonse on the basis of a client-provided `latest_seen` timestamp,
+//! i.e., it will return all rapid gossip sync data it has seen after the given timestamp.
+//!
+//! # Getting Started
+//! Firstly, the data needs to be retrieved from the server. For example, you could use the server
```
[the remark this answers is not in the document]
Comment (tnull) [reply lightning-rapid-gossip-sync/src/lib.rs:31]: We do so in the following sentence? Reads fine as is to me?

File: lightning-rapid-gossip-sync/src/lib.rs [hunk 6]
```diff
@@ -10,24 +10,36 @@
...
+//! at <https://rapidsync.lightningdevkit.org> with the following request format:
 //!
 //! ```shell
 //! curl -o rapid_sync.lngossip https://rapidsync.lightningdevkit.org/snapshot/<last_sync_timestamp>
 //! ```
+//! Note that the
```
[the remark this answers is not in the document]
Comment (tnull) [reply lightning-rapid-gossip-sync/src/lib.rs:37]: No idea. :shrug:

File: lightning-rapid-gossip-sync/src/lib.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (arik-so) [reply lightning-rapid-gossip-sync/src/lib.rs:31]: well, we don't use a pronoun in the followup sentence, so it doesn't feel as direct, but this comment was just a question and not a correction :)

**Review by arik-so — DISMISSED** [review]
lgtm

**Comment by tnull** [comment]
I took the liberty to squash fixup commits.

**Comment by tnull** [comment]
> LGTM. If you're bored, #1566 is easy :p

Will have a look!

File: lightning-rapid-gossip-sync/src/lib.rs [hunk 7]
```diff
@@ -75,34 +72,31 @@ use std::sync::atomic::{AtomicBool, Ordering};
...
+mod error;
 
 /// Core functionality of this crate
-pub mod processing;
+mod processing;
 
-/// Rapid Gossip Sync struct
-/// See [crate-level documentation] for usage.
-///
-/// [crate-level documentation]: crate
```
[the remark this answers is not in the document]
Comment (tnull) [reply lightning-rapid-gossip-sync/src/lib.rs:89]: Because it also showed in the crate-level docs itself, and it seemed weird to have a loop there?

**Comment by tnull** [comment]
> No opinion. You do you.

Alright, then I'll here, probably tomorrow though.

File: lightning-rapid-gossip-sync/src/lib.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (tnull) [reply lightning-rapid-gossip-sync/src/lib.rs:89]: Re-added the link. Turns out it was only missing one linebreak.

**Comment by tnull** [comment]
> LGTM, feel free to squash.

Squashed!

**Review by arik-so — APPROVED** [review]

**Pull request merged**